ci/docker/base.Dockerfile

```
@@ -82,4 +82,4 @@ RUN pyenv update && \
 ENV PATH="/root/.pyenv/shims:${PATH}"
 
 ARG CUPY_PACKAGE=cupy-cuda11x
-RUN pip install --upgrade pip setuptools wheel tox clang-format ${CUPY_PACKAGE}
+RUN pip install --upgrade pip setuptools wheel uv nox clang-format ${CUPY_PACKAGE}
```

```suggestion
RUN pip install --upgrade uv nox
```
Probably that's the only thing that is useful here, because later we create a venv, which will not contain the other stuff.
Therefore, we don't have cupy in the environment, therefore the gpu test in `tests/py2fgen/test_cli.py` fails. Not sure what's the best way to define the cupy package version and then install it with nox/uv.

---

Some more cleanup can be applied to the Dockerfile, but as discussed with @egparedes it can be done after this PR is merged.

For example, the `pyenv` is probably not needed and these lines can be removed:
```
ENV PYENV_ROOT /root/.pyenv
ENV PATH="/root/.pyenv/bin:${PATH}"

RUN pyenv update && \
    pyenv install ${PYVERSION} && \
    echo 'eval "$(pyenv init -)"' >> /root/.bashrc && \
    eval "$(pyenv init -)" && \
    pyenv global ${PYVERSION}

ENV PATH="/root/.pyenv/shims:${PATH}"
```

Instead, we could just install `uv` with:
```
RUN curl -LsSf https://astral.sh/uv/install.sh | sh
ENV PATH="/root/.local/bin:${PATH}"
```